x-pack/legacy/plugins/apm/public/components/app/ServiceDetails/ServiceDetailTabs.tsx

```diff
@@ -33,20 +38,31 @@ export function ServiceDetailTabs({ urlParams }: Props) {
     }
   }, [serviceName, start, end]);
 
+  if (!serviceName) {
+    // this never happens, urlParams type is not accurate enough
```

Comment:
Thanks for adding this. We can probably solve this later with some io-ts at the routing boundary.